src/main/java/network/brightspots/rcv/CastVoteRecord.java

```diff
@@ -44,6 +44,12 @@
   private final String suppliedId;
   // which precinct this ballot came from
   private final String precinct;
+  // contest associated with this CVR
+  private String contestId = null;
```

Comment:
Why aren't these new Strings final? Also, I think giving them default values of null happens automatically.